project/ianlienfa/README.rst

```diff
@@ -0,0 +1,122 @@
+3. Industry professionals that would like to try out some ideas before moving onto implementing an state-of-the-art solver
+
+System Architecture
+===================
+
+`BB_engine`: 
+  This is the main engine of the framework.  It is responsible for the following:
+  1. Read the input file and parse the input
+  2. Construct the environment
+  3. Run the search strategy
+  4. Output the results
+
+`branch_modules`:
+  This is the collection of branch classes
+  A branch class should inherit the base class `BranchMod`, which requires the following:
+  * Implement the `branch` function
+
+`prune_modules`:
+  This is the collection of pruning classes
+  A pruning class should inherit the base class `PruneMod`, which requires the following:
+  * Implement the `prune` function
+
+`Node`:
+  This is the collection of node classes
+  A specific problem requires a node class and a graph class to be defined.
+  It generically specifies the node information during branch and bound
+
+`Graph`:
+  The graph class is used to store the graph information during branch and bound
+  Compared to the node class, the graph class has more global information and knows about the state of the search
+  For example, a graph class for the :math: `1|\Sigma_Cj` problem, we need:
+    * the configuration of the probelm instance
+    * the best value founded so far    
+    * the data structure for searching
+
+`config.h`:
+  The configuration for the problem solving
+  Users can twiddle with the configuration in this file, 
+  including the problem type, the searching strategy, the branching strategy, and the pruning strategy
+
+In general, the framework calls the BB_engine at the start of the program, 
+then for every node visited, the framework calls the branch_modules to branch, 
+the prune_modules to prune, updates the graph, then relies on the search_module to search for the next node.
+
+The users can use this system solely from source, twiddle the configuration in the config.h file, and build them from the source.  
+For this project, we hope to wrap it in a python package, and provide a gym-like api for RL on branch and bound problem solving.
+
+
+API Description
+===============
+
+The following API follows the OpenAI gym api for environments
+The basic usage will be like following:
+
+```python
+import gym
+env = gym.make('BBGym-v0')
+done = False
+while not done:
+  action = env.action_space.sample()
+  observation, reward, done, info = env.step(action)
+  print(observation, reward, done, info)
+
+```
+
+Engineering Infrastructure
+==========================
+
+Build System:
+This project uses cmake for its build system, git as version control system
+and uses google test and pytest as testing framework.
+
+The documentation will be presented on github 
+
+
+Schedule
+========
+
+* Planning phase (6 weeks including 2/21, 2/28, 3/7, 3/14, 3/21, 3/28):
+* Week 1 (4/4): 
+    * Testing environment setup
+* Week 2 (4/11):
+    * Pybind binding for the framework
```

Comment:
Eventually you may have the binding and the "framework adjustment" work mingled together.  Perhaps you can try to list the sequence of development for all your modules.

Reply:
Schedule updated